### PR TITLE
8260257: [Linux] WebView no longer reacts to some mouse events

### PR DESCRIPTION
**Pull request by arun-Joseph**
Timer in RunLoopGeneric has an open bug in WebKit (https://bugs.webkit.org/show_bug.cgi?id=189335) causing the timer to remain active even after firing.

Reverting back to WebCore Timer for ScrollAnimation in Linux.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8260257](https://bugs.openjdk.java.net/browse/JDK-8260257): [Linux] WebView no longer reacts to some mouse events

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Guru Hb](https://openjdk.java.net/census#ghb) (@guruhb - **Reviewer**) 🔄 Re-review required (review applies to 480c0d98672c90588eb426c03d08ee8b9707f394)
 * [Johan Vos](https://openjdk.java.net/census#jvos) (@johanvos - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jfx pull/404/head:pull/404`
`$ git checkout pull/404`

### CODE REVIEW

**Review by kevinrushforth — APPROVED** [review]
Looks good. Tested on all three platforms (although it should only affect Linux). I confirm that on Linux scrolling via the trackpad (or by clicking in the scrollbar track) now works properly.

**Review by guruhb — APPROVED** [review]
Looks good to me. Verified on Ubuntu 20.04

**Comment by kevinrushforth** [comment]
As discussed offline, an automated test would be helpful to ensure no future regression.

**Review by johanvos — APPROVED** [review]
Great.
The test fails before and succeeds after the patch.

File: tests/system/src/test/java/test/javafx/scene/web/WebPageTest.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                if (newValue) {
+                    webViewStateLatch.countDown();
+                }
+            });
+
+            webView.getEngine().loadContent(htmlContent);
+        });
+
+        assertTrue("Timeout when waiting for focus change ", Util.await(webViewStateLatch));
+
```
Comment (kevinrushforth) [inline tests/system/src/test/java/test/javafx/scene/web/WebPageTest.java:135]: I needed to add a `sleep(500)` here.

File: tests/system/src/test/java/test/javafx/scene/web/WebPageTest.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        Util.runAndWait(() -> {
+            final WebPage page = WebEngineShim.getPage(webView.getEngine());
+            assertNotNull(page);
+            WebPageShim.scroll(page, 1, 1, 0, 100);
+        });
+
+        Util.sleep(100);
```
Comment (kevinrushforth) [inline tests/system/src/test/java/test/javafx/scene/web/WebPageTest.java:142]: I needed to change this to `sleep(500)`

**Review by kevinrushforth — COMMENTED** [review]
The new test looks good, although in order to make it pass on my slower Ubuntu 20.04 VM, I had to increase the sleep time to 500 ms and also add a sleep after the focus latch and before the scroll. See below.

**Comment by kevinrushforth** [comment]
I did a little more experimenting, and the important one on my Linux VM system is the addition of the 500 ms sleep between the await on the latch and initiating the scroll. I could leave the other delay at 100 ms and it still passes for me. It may be best to increase that latter sleep to 500 anyway.

File: tests/system/src/test/java/test/javafx/scene/web/WebPageTest.java (hunk 1 above)
Comment (arun-Joseph) [reply tests/system/src/test/java/test/javafx/scene/web/WebPageTest.java:136]: I've used `Util.sleep(1000)` instead to make it similar to the other system web tests.

**Review by kevinrushforth — APPROVED** [review]

**Review by johanvos — APPROVED** [review]

**Pull request closed without merging**